### ISSUE TITLE
English-Sanskrit Dictionaries Punctuation Mismarkup

### ISSUE THREAD

**Issue by gasyoun** [sanskrit-lexicon/csl-corrections#13]
`, ; .` mismarked as part of Sanskrit words

https://www.sanskrit-lexicon.uni-koeln.de/scans/MWEScan/2020/web/webtc/indexcaller.php

`3) CAVE or CAVERN, s. vilaṃ, vivaraṃ, kandaraḥ -rā, garttaḥ, gahvaraṃ -rī, randhraṃ,
guhā, guhāgṛhaṃ, nirddariḥ m., darī f., śailarandhraṃ. — (Excavation) khātaṃ,
devakhātaṃ, devakhātavilaṃ.`

![mw en](https://user-images.githubusercontent.com/80761/75660062-297d5080-5c7c-11ea-81c1-09ddb6275eb9.PNG)

Same in

`CAVE, CAVERN: I Natural: (1) guhā, entered
the great c. praviveśa mahāguhām, V. p.; (2) gahvaram;
(3) kandaraḥ (rī, raṃ); (4) darī or dariḥ. II Den,
c a y: q. v. `

![caved](https://user-images.githubusercontent.com/80761/75660497-f091ab80-5c7c-11ea-96b9-644e86c7e295.PNG)

and

`Cave, Cavern, s. guhā, darī, gahvaraṃ, kaṃdaraḥ-
-rā, vivaraḥ-raṃ. `

![aptes](https://user-images.githubusercontent.com/80761/75660673-1028d400-5c7d-11ea-9bca-397828e18e8d.PNG)

**Comment by funderburkjim** [comment]
Here is the current digitization (from ae.txt) for 'Cave':
```
{@Cave, Cavern,@}¦ {%s.%} {#guhA, darI, gahvaraM, kaMdaraH-#}
<div n="lb"/>{#-rA, vivaraH-raM#}.
```

And, here is the constructed xml (ae.xml):
```
<b>Cave, Cavern,</b>  <i>s.</i> <s>guhA, darI, gahvaraM, kaMdaraH-</s> <
div n="lb"/><s>-rA, vivaraH-raM</s>.
```

Questions 
* 1:  Do you care just about the display (which is based on ae.xml)?
* 2: Why does it matter to you?

**Comment by gasyoun** [comment]
>  Do you care just about the display (which is based on ae.xml)?

No, about the structure. I would prefer to have the punctuation excluded.

> Why does it matter to you?

I'm thinking of a Russian-Sanskrit dictionary as well. But words written as now are in many cases pieces of words, not real words.

**Comment by funderburkjim** [comment]
This would be a big task, to separate out the punctuation.

In terms of degree of difficulty, maybe there are two cases:
* comma, semicolon, period.   These would be changeable by regular expression
   substitutions.  e.g. 
   `{#guhA, darI, gahvaraM, kaMdaraH-#}` -> 
   `{#guhA#}, {#darI#}, {#gahvaraM#}, {#kaMdaraH-#}'
* hyphens are not straightforward. 
   `{#kaMdaraH-#} {#-rA, vivaraH-raM#}`. fully expanded might be 
   `{#kaMdaraH#}, {#kaMdarA#}, {#vivaraH#}, {#vivaraM#}.`
   Probably a program could generate most such expansions involving '-',
   but note that the expansion would greatly differ from the digitization.
   I am dubious about changing the digitization so much.

The underlying concern is probably how to be able to search for 'kaMdarA' or
  'vivaraM' and get the 'Cave' entry as part of the search result.

**Comment by gasyoun** [comment]
> This would be a big task, to separate out the punctuation.

Could it be something we could reuse in the rest of the dictionarys as well?

> comma, semicolon, period. These would be changeable by regular expression
substitutions

Let's go for the low hanging fruits?

> but note that the expansion would greatly differ from the digitization.

Indeed, but as it is, the list can't be used.

> I am dubious about changing the digitization so much.

Maybe we should leave it as it is and add a new layer?

> The underlying concern is probably how to be able to search for 'kaMdarA' or
'vivaraM' and get the 'Cave' entry as part of the search result.

Exactly.

**Comment by funderburkjim** [comment]
> add a new layer

For dictionaries where Thomas' digitization closely followed the line breaks of the print version, a new layer  might be the way to go.  
Current work flow: ae.txt -> ae.xml -> displays,
New replacement work flow: ae.txt -> ae1.txt -> ae.xml -> displays

A (rather complicated, no doubt) python program would do the ae.txt -> ae1.txt transformation.

This is just thinking out loud, very preliminary.  But it is nice to contemplate a version of,
say ae, that would keep the spirit and data of the original 1884 work, and that would also reduce print edition obstacles.

**Comment by gasyoun** [comment]
>  reduce print edition obstacles.

Yes, because now they make the data non-reusable. 

> spirit and data of the original 1884 work

Spirit remains for sure. It's there and not at the same time.

> For dictionaries where Thomas' digitization closely followed the line breaks of the print version, a new layer might be the way to go.

Yeah, and it is not as much about the line brakes, as the printer's conventions, I guess.

**Comment by drdhaval2785** [comment]
This issue is resolved partially. Terminal commas, colons, semicolons are out of Devanagari markup now. The other part (non-terminal) is being handled at #178 if required. Closing this old issue
